### ISSUE TITLE
Quick switch

### ISSUE THREAD

**Issue by mottosso** [mindbender-studio/launcher#11]
### Goal

Save time browsing for assets.

### Implemenation

Mimic the Windows Explorer mechanism of browsing to an arbitrary level in any of the currently selected levels.

![image](https://cloud.githubusercontent.com/assets/2152766/25556019/9502e15c-2cec-11e7-869c-4ec21950f815.png)

**Comment by BigRoy** [comment]
Does this mean clicking on "assets" and directly going there? Or do you mean the pop-up list when clicking on the arrows in-between?

I'm looking for the first actually and tried adding `index` in the `controller.pop()` for the Breadcrumbs.qml (which is a QML Repeater) but somehow the index is always `1`. Any idea how to pass to Python what index or button is pressed of the repeater?

**Comment by mottosso** [comment]
I mean both, but the former is a good first step.

As for how to go about it, if you mean this [`controller.pop()`](https://github.com/getavalon/launcher/blob/921c2764228f8da43cf60dc4b907065b4534979b/launcher/res/qml/Breadcrumbs.qml#L44) then I think you're on the right track. But it might not be an integer, but a QModelIndex. You'll need to tell the slot to accept QModelIndex [here](https://github.com/getavalon/launcher/blob/921c2764228f8da43cf60dc4b907065b4534979b/launcher/control.py#L375) as well. Then you can get the integer index via `index.row()`.

**Comment by BigRoy** [comment]
According to [this page](http://doc.qt.io/qt-5/qml-qtquick-repeater.html) the following would show it just as integers:

```
Column {
    Repeater {
        model: 10
        Text { text: "I'm item " + index }
    }
}
```

So I'd expect them to be integers. I was able to have it also print the index, it would just always be the integer `1` somehow.

**Comment by mottosso** [comment]
Possibly. In that case, it should work.

**Comment by BigRoy** [comment]
> Possibly. In that case, it should work.

That's what I assumed, but for each button - no matter which I clicked it - it was giving me a value of one. Maybe I'll have another stab at this soon. :)

**Comment by BigRoy** [comment]
Regarding this: https://github.com/getavalon/launcher/blob/921c2764228f8da43cf60dc4b907065b4534979b/launcher/res/qml/Breadcrumbs.qml#L44

Ok, so I think I see what the issue is. The `index` in that scope in the QML file is probably from the Delegate. I need to somehow bring along (in QML) the index from the scope above it (the Repeater) into the delegate (the Toolbox).

**Comment by mottosso** [comment]
Hm, the delegate is a ToolButton, and it's got no `index` member. Try `model.index` to explicitly refer to the delegate's model.

**Comment by BigRoy** [comment]
Actually. @aardschok is working on this now and we're getting to something functional now. I must have missed adding `int` to the `Slot` when I tried this before.

Cool stuff coming up soon.

**Comment by aardschok** [comment]
Solution was to give the `Slot` decorator  and the method `controller.pop` an `int` argument. In the `Breadcrumbs` qml file we needed to give the `index` to `controller.pop` of the repeater.

See https://github.com/getavalon/launcher/pull/22

**Comment by ruchitinfushion** [comment]
Right now breadcrumbs navbar can jump only to current index -1, right ?
Let say user trail is " batman > assets > bruce > modeling", After this if user click on batman then QListView will set to "anim, layout, lookdev,etc" instead of "assets,film".

**Comment by tokejepsen** [comment]
Pretty sure the quick switch has been implemented?

**Comment by BigRoy** [comment]
Yes, should be with: https://github.com/getavalon/launcher/pull/22

We have it working on our Launcher at least, and I believe it's merged into here too.

**Comment by davidlatwe** [comment]
Supplying evidence GIF

![launcher_quick_switch](https://user-images.githubusercontent.com/3357009/47641665-9de44300-dba1-11e8-9bc3-386d0c74575c.gif)

:beers:
